api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1713,7 +1714,7 @@ public ObjectStat statObject(StatObjectArgs args)
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
     checkArgs(args);
-    checkReadRequestSse(args.ssec());
+    args.validateSsec(baseUrl);
```

Comment:
Is it a good idea to have this check inside `checkArgs()`?

Reply:
`checkArgs` accepts the base class `BaseArgs`, so the `validateSsec()` method which is defined in `SsecObjectArgs` won't be available there.